src/CheckBox/index.jsx

```diff
@@ -1,69 +1,62 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-import { inputProps } from '../asInput';
+import asInput from '../asInput';
 import newId from '../utils/newId';
 
-class CheckBox extends React.Component {
+class Check extends React.Component {
   constructor(props) {
     super(props);
 
-    this.handleClick = this.handleClick.bind(this);
-    if (this.props.onChange) {
-      this.onChange = this.props.onChange.bind(this);
-    }
+    this.onChange = this.onChange.bind(this);
 
     const id = newId('checkbox');
-    if (props.checked === 'true') {
-      this.state = {
-        id,
-        checked: true,
-      };
-    } else {
-      this.state = {
-        id,
-        checked: false,
-      };
-    }
+    this.state = {
+      id,
+      checked: props.checked || false,
+    };
   }
 
-  handleClick() {
-    if (this.state.checked === true) {
-      this.setState({
-        checked: false,
-      });
-    } else {
-      this.setState({
-        checked: true,
-      });
-    }
-    if (this.onChange) {
-      this.onChange();
-    }
+  onChange(event) {
+    this.setState({
+      checked: !this.state.checked,
+    });
+    this.props.onChange(event);
   }
 
 
   render() {
     const props = { ...this.props };
 
     return (
-      <label htmlFor={this.state.id}>
-        <input
-          id={this.state.id}
-          name={props.name}
-          type="checkbox"
-          defaultChecked={this.state.checked}
-          aria-describedby={props.describedBy}
-          aria-checked={this.state.checked}
-          tabIndex="0"
-          onClick={this.handleClick}
-          disabled={props.disabled}
-        />
-        {props.label}
-      </label>
+      <input
+        id={props.id}
+        type="checkbox"
+        name={props.name}
+        defaultChecked={this.state.checked}
+        aria-checked={this.state.checked}
+        onChange={this.onChange}
+        disabled={props.disabled}
+      />
     );
   }
 }
 
-CheckBox.propTypes = inputProps;
+Check.propTypes = {
+  checked: PropTypes.bool,
+  onChange: PropTypes.func,
+};
+
+Check.defaultProps = {
+  checked: false,
+  onChange: () => {},
+};
+
+const CheckBox = asInput(Check, false);
+
+CheckBox.propTypes = {
```

Comment:
You should be able to remove this propTypes entirely. The propTypes and defaultProps above aren't ideal, but that's not your fault and it's something I can solve in asInput later on.